### ISSUE TITLE
Running tests sequentially

### ISSUE THREAD

**Issue by anoushkadoctor** [microsoft/playwright-test#235]
I have about 13 test suites (.spec.ts) files and i want to run all 13 files in a specific order each time. Only after one test suite has ended, should a new one begin. Is there a way to achieve this?

All of my test suites have the following structure:
```
import {expect, it, describe} from './fixtures'
describe("something", ()=>{
        it("test1", async({page, data})=>{
        });
        .
        .
        .
        //other it blocks...
});
```

**Comment by srguiwiz** [comment]
I was looking at folio's `cli.js` and in
``` javascript
async function collectFiles(testDir) {
    const list = [];
    let callback;
    const result = new Promise(f => callback = f);
    fstream_ignore_1.default({ path: testDir, ignoreFiles: ['.gitignore'] })
        .on('child', (c) => list.push(c.path))
        .on('end', () => callback(list.sort()));
    return result;
}
```
I changed
``` javascript
        .on('end', () => callback(list));
```
to
``` javascript
        .on('end', () => callback(list.sort()));
```
We still didn't get the results I expected.

But, it is around there, in `collectFiles` and `filterFiles` and `runTests`.

I still look for a way to run tests in order too.  Willing to settle for numbering, alphabetic order, or multiple `--test-match` etc..

**Comment by mxschmitt** [comment]
Tests should be independent from each other and not depend on other tests or test files. If you need logic before and after it's recommended to have it inside each test or use beforeEach/beforeAll to prepare a login e.g. Also playwright-test already has a new context each test, so its a "fresh state with no cookies, localStorage etc)

Why do you want to have them in a custom order in the first place?

**Comment by srguiwiz** [comment]
Two reasons to run in order:

One, nice visual demo.

Two, I was trying to find out whether certain information leaks from one to the other, e.g. folio.config properties.  Someone saying so may or may not be true, so I wanted to verify.  Then I noticed I was not able to control order of execution.

**Comment by anoushkadoctor** [comment]
@mxschmitt Agreed that tests should not depend on each other. However we do have some dependencies on the data that the test puts into our system. In that case, how would you suggest using beforeEach/beforeAll to perform a test? 
Eg: I need X.spec.ts and Y.spec.ts to run before Z.spec.ts. How would we achieve this using beforeAll/beforeEach?
